### ISSUE TITLE
Error in r[cbind(1L:p, 1L:p)] <- 1 : subscript out of bounds

### ISSUE THREAD

**Issue by huipan1973** [GabrielHoffman/variancePartition#11]
Hi,

I have the error  "Error in r[cbind(1L:p, 1L:p)] <- 1 : subscript out of bounds" when I am running dream.

> traceback()
6: cov2cor(C)
5: .local(object, ...)
4: classifyTestsF(out[, realcoef], df = df, fstat.only = TRUE)
3: classifyTestsF(out[, realcoef], df = df, fstat.only = TRUE)
2: .standard_transform(ret)
1: dream(expr, formula = fm1, data = dat, L = L)

> sessionInfo()
R version 3.6.2 (2019-12-12)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] variancePartition_1.16.1 Biobase_2.46.0           BiocGenerics_0.32.0      scales_1.1.0            
[5] foreach_1.4.8            ggplot2_3.2.1            limma_3.42.2            

loaded via a namespace (and not attached):
 [1] progress_1.2.2      gtools_3.8.1        tidyselect_1.0.0    purrr_0.3.3         reshape2_1.4.3     
 [6] lmerTest_3.1-1      splines_3.6.2       lattice_0.20-38     colorspace_1.4-1    vctrs_0.2.2        
[11] snow_0.4-3          rlang_0.4.4         pillar_1.4.3        nloptr_1.2.1        glue_1.3.1         
[16] withr_2.1.2         BiocParallel_1.20.1 lifecycle_0.1.0     plyr_1.8.5          stringr_1.4.0      
[21] munsell_0.5.0       gtable_0.3.0        caTools_1.18.0      codetools_0.2-16    doParallel_1.0.15  
[26] pbkrtest_0.4-8.6    Rcpp_1.0.3          KernSmooth_2.23-16  gdata_2.18.0        lme4_1.1-21        
[31] gplots_3.0.1.2      hms_0.5.3           stringi_1.4.6       dplyr_0.8.4         numDeriv_2016.8-1.1
[36] grid_3.6.2          tools_3.6.2         bitops_1.0-6        magrittr_1.5        lazyeval_0.2.2     
[41] tibble_2.1.3        crayon_1.3.4        pkgconfig_2.0.3     MASS_7.3-51.4       Matrix_1.2-18      
[46] prettyunits_1.1.1   assertthat_0.2.1    minqa_1.2.4         rstudioapi_0.11     iterators_1.0.12   
[51] R6_2.4.1            boot_1.3-23         colorRamps_2.3      nlme_3.1-142        compiler_3.6.2

**Comment by GabrielHoffman** [comment]
Can you show a minimal reproducible example that produces the error

- Gabriel

Sent from my iPhone

On Feb 23, 2020, at 8:52 PM, Hui Pan <notifications@github.com> wrote:

﻿
USE CAUTION: External Message.

Hi,

I have the error "Error in r[cbind(1L:p, 1L:p)] <- 1 : subscript out of bounds" when I am running dream.

traceback()
6: cov2cor(C)
5: .local(object, ...)
4: classifyTestsF(out[, realcoef], df = df, fstat.only = TRUE)
3: classifyTestsF(out[, realcoef], df = df, fstat.only = TRUE)
2: .standard_transform(ret)
1: dream(expr, formula = fm1, data = dat, L = L)

sessionInfo()
R version 3.6.2 (2019-12-12)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 LC_CTYPE=English_United States.1252
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C
[5] LC_TIME=English_United States.1252

attached base packages:
[1] parallel stats graphics grDevices utils datasets methods base

other attached packages:
[1] variancePartition_1.16.1 Biobase_2.46.0 BiocGenerics_0.32.0 scales_1.1.0
[5] foreach_1.4.8 ggplot2_3.2.1 limma_3.42.2

loaded via a namespace (and not attached):
[1] progress_1.2.2 gtools_3.8.1 tidyselect_1.0.0 purrr_0.3.3 reshape2_1.4.3
[6] lmerTest_3.1-1 splines_3.6.2 lattice_0.20-38 colorspace_1.4-1 vctrs_0.2.2
[11] snow_0.4-3 rlang_0.4.4 pillar_1.4.3 nloptr_1.2.1 glue_1.3.1
[16] withr_2.1.2 BiocParallel_1.20.1 lifecycle_0.1.0 plyr_1.8.5 stringr_1.4.0
[21] munsell_0.5.0 gtable_0.3.0 caTools_1.18.0 codetools_0.2-16 doParallel_1.0.15
[26] pbkrtest_0.4-8.6 Rcpp_1.0.3 KernSmooth_2.23-16 gdata_2.18.0 lme4_1.1-21
[31] gplots_3.0.1.2<https://urldefense.proofpoint.com/v2/url?u=http-3A__3.0.1.2&d=DwQCaQ&c=shNJtf5dKgNcPZ6Yh64b-A&r=KdYcmw5SdXylMrTGSuNVkNJulowod64k0PTDC5BHZkk&m=8pLg5MVgqW2Jj6xlCYM2SumOC3FGk3Lo86HN8GcSFc4&s=FDA39IDSnTaGyDkIezObJy3h-NWkBg43NZS_K6gnQQk&e=> hms_0.5.3 stringi_1.4.6 dplyr_0.8.4 numDeriv_2016.8-1.1
[36] grid_3.6.2 tools_3.6.2 bitops_1.0-6 magrittr_1.5 lazyeval_0.2.2
[41] tibble_2.1.3 crayon_1.3.4 pkgconfig_2.0.3 MASS_7.3-51.4 Matrix_1.2-18
[46] prettyunits_1.1.1 assertthat_0.2.1 minqa_1.2.4 rstudioapi_0.11 iterators_1.0.12
[51] R6_2.4.1 boot_1.3-23 colorRamps_2.3 nlme_3.1-142 compiler_3.6.2

**Comment by huipan1973** [comment]
Hi Gabriel,

Here is an example. However, it's fine when I run it on my work computer but not my home computer.
I install variancePartition two days ago on my home computer. Is that the reason?

Regards,
Hui

_________________________________________________________________________
library(limma)
library(variancePartition)
set.seed(100)
a <- 1:7 + rep(c(0,2,5), time = c(3,2,2)) + rnorm(7, sd = 3)
b <- (1:7)+ sample(seq(0.1, 3, by = 0.1), 7) + rep(c(0,2,5), time = c(3,2,2)) + rnorm(7, sd = 3)
dat <- data.frame(AB1 = c(a, b),
                  AB2 = c(a, b) + rnorm(14, sd = 1),
                  AB3 = c(a, b) + rnorm(14, sd = 3),
                  AB4 = c(a, b) + rnorm(14, sd = 5),
                  AB5 = c(a, b) + rnorm(14, sd = 10),
                  GROUP = rep(c("a", "b"), each = 7),
                  ID = rep(paste0("id", 1:7), 2),
                  BATCH = rep(rep(c("b1", "b2", "b3"), time = c(3,2,2)), 2))
rownames(dat) <- paste0("samp", 1:14)
expr <- t(dat[, c("AB1", "AB2", "AB3", "AB4", "AB5")])

# fit linear mixed model
fm1 <- ~ GROUP + BATCH + (1|ID)
L = getContrast(expr, formula = fm1, data = dat, coefficient = c("GROUPb"))
fit1 <- dream(expr, formula = fm1, data = dat, L = L)
__________________________________________________________________________
________________________________

**Comment by huipan1973** [comment]
Hi Gabriel,
I would also like to hear your advice on how to use limma eBayes after dream, or why I can't use it.
Regards,Hui
    On Monday, February 24, 2020, 07:53:02 AM EST, Hui Pan <hui_pan@outlook.com> wrote:  
 
 Hi Gabriel,
Here is an example. However, it's fine when I run it on my work computer but not my home computer.I install variancePartition two days ago on my home computer. Is that the reason?
Regards,Hui
_________________________________________________________________________library(limma)
library(variancePartition)
set.seed(100)
a <- 1:7 + rep(c(0,2,5), time = c(3,2,2)) + rnorm(7, sd = 3)
b <- (1:7)+ sample(seq(0.1, 3, by = 0.1), 7) + rep(c(0,2,5), time = c(3,2,2)) + rnorm(7, sd = 3)
dat <- data.frame(AB1 = c(a, b), 
                  AB2 = c(a, b) + rnorm(14, sd = 1),
                  AB3 = c(a, b) + rnorm(14, sd = 3), 
                  AB4 = c(a, b) + rnorm(14, sd = 5),
                  AB5 = c(a, b) + rnorm(14, sd = 10),
                  GROUP = rep(c("a", "b"), each = 7),
                  ID = rep(paste0("id", 1:7), 2),
                  BATCH = rep(rep(c("b1", "b2", "b3"), time = c(3,2,2)), 2))
rownames(dat) <- paste0("samp", 1:14)
expr <- t(dat[, c("AB1", "AB2", "AB3", "AB4", "AB5")])

# fit linear mixed model
fm1 <- ~ GROUP + BATCH + (1|ID)   
L = getContrast(expr, formula = fm1, data = dat, coefficient = c("GROUPb"))
fit1 <- dream(expr, formula = fm1, data = dat, L = L)  
__________________________________________________________________________From: Gabriel Hoffman <notifications@github.com>
Sent: Sunday, February 23, 2020 9:29 PM
To: GabrielHoffman/variancePartition <variancePartition@noreply.github.com>
Cc: Hui Pan <hui_pan@outlook.com>; Author <author@noreply.github.com>
Subject: Re: [GabrielHoffman/variancePartition] Error in r[cbind(1L:p, 1L:p)] <- 1 : subscript out of bounds (#11) Can you show a minimal reproducible example that produces the error 

- Gabriel 

Sent from my iPhone 

On Feb 23, 2020, at 8:52 PM, Hui Pan <notifications@github.com> wrote: 

﻿ 
USE CAUTION: External Message. 

Hi, 

I have the error "Error in r[cbind(1L:p, 1L:p)] <- 1 : subscript out of bounds" when I am running dream.

traceback() 
6: cov2cor(C) 
5: .local(object, ...) 
4: classifyTestsF(out[, realcoef], df = df, fstat.only = TRUE) 
3: classifyTestsF(out[, realcoef], df = df, fstat.only = TRUE) 
2: .standard_transform(ret) 
1: dream(expr, formula = fm1, data = dat, L = L) 

sessionInfo() 
R version 3.6.2 (2019-12-12) 
Platform: x86_64-w64-mingw32/x64 (64-bit) 
Running under: Windows 7 x64 (build 7601) Service Pack 1 

Matrix products: default 

locale: 
[1] LC_COLLATE=English_United States.1252 LC_CTYPE=English_United States.1252 
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C 
[5] LC_TIME=English_United States.1252 

attached base packages: 
[1] parallel stats graphics grDevices utils datasets methods base 

other attached packages: 
[1] variancePartition_1.16.1 Biobase_2.46.0 BiocGenerics_0.32.0 scales_1.1.0 
[5] foreach_1.4.8 ggplot2_3.2.1 limma_3.42.2 

loaded via a namespace (and not attached): 
[1] progress_1.2.2 gtools_3.8.1 tidyselect_1.0.0 purrr_0.3.3 reshape2_1.4.3 
[6] lmerTest_3.1-1 splines_3.6.2 lattice_0.20-38 colorspace_1.4-1 vctrs_0.2.2 
[11] snow_0.4-3 rlang_0.4.4 pillar_1.4.3 nloptr_1.2.1 glue_1.3.1 
[16] withr_2.1.2 BiocParallel_1.20.1 lifecycle_0.1.0 plyr_1.8.5 stringr_1.4.0 
[21] munsell_0.5.0 gtable_0.3.0 caTools_1.18.0 codetools_0.2-16 doParallel_1.0.15
[26] pbkrtest_0.4-8.6 Rcpp_1.0.3 KernSmooth_2.23-16 gdata_2.18.0 lme4_1.1-21 
[31] gplots_3.0.1.2<https://urldefense.proofpoint.com/v2/url?u=http-3A__3.0.1.2&d=DwQCaQ&c=shNJtf5dKgNcPZ6Yh64b-A&r=KdYcmw5SdXylMrTGSuNVkNJulowod64k0PTDC5BHZkk&m=8pLg5MVgqW2Jj6xlCYM2SumOC3FGk3Lo86HN8GcSFc4&s=FDA39IDSnTaGyDkIezObJy3h-NWkBg43NZS_K6gnQQk&e=> hms_0.5.3 stringi_1.4.6 dplyr_0.8.4 numDeriv_2016.8-1.1 
[36] grid_3.6.2 tools_3.6.2 bitops_1.0-6 magrittr_1.5 lazyeval_0.2.2 
[41] tibble_2.1.3 crayon_1.3.4 pkgconfig_2.0.3 MASS_7.3-51.4 Matrix_1.2-18 
[46] prettyunits_1.1.1 assertthat_0.2.1 minqa_1.2.4 rstudioapi_0.11 iterators_1.0.12
[51] R6_2.4.1 boot_1.3-23 colorRamps_2.3 nlme_3.1-142 compiler_3.6.2

**Comment by GabrielHoffman** [comment]
1) Thanks for the code example.  Unfortunately, I was not able to reproduce the error on multiple version of variancePartition on multiple versions of R on OS X, Windows and CentOS.  I can't fix this issue unless I can reproduce it.

2) I originally included an `eBayes()` step in the original version of dream.  However, based on:
  
   1. a better understanding of the underlying theory as applied to linear mixed models, 
   2. issued with empirical performance in simulated data for more complex study designs,     
   3. feedback from users and reviewers,  

I removed the `eBayes()` step from the current version. 

I am currently thinking about how to include a more robust, and theoretically motivated `eBayes()` step for linear mixed models in the future, but it isn't ready yet.

**Comment by huipan1973** [comment]
Thanks for the explanation.
Regards,Hui

Sent from Yahoo Mail on Android 
 
  On Mon, Feb 24, 2020 at 2:04 PM, Gabriel Hoffman<notifications@github.com> wrote:      
   -    
Thanks for the code example. Unfortunately, I was not able to reproduce the error on multiple version of variancePartition on multiple versions of R on OS X, Windows and CentOS. I can't fix this issue unless I can reproduce it.

   -    
I originally included an eBayes() step in the original version of dream. However, based on:
      
      - a better understanding of the underlying theory as applied to linear mixed models,
      - issued with empirical performance in simulated data for more complex study designs,
      - feedback from users and reviewers,


I removed the eBayes() step from the current version.

I am currently thinking about how to include a more robust, and theoretically motivated Bayes() step for linear mixed models in the future, but it isn't ready yet.

**Comment by raungar** [comment]
Hi - 

I get this error as well all of the sudden from my home computer (mac 10.14.4) running simply your dream analysis section of the vignette (https://www.bioconductor.org/packages/devel/bioc/vignettes/variancePartition/inst/doc/dream.html).  . This hasn't happened before, maybe from a recent update? It works on my cluster however as well. 


My home computer sessionInfo()
```

R version 3.6.3 (2020-02-29)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.4

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

Random number generation:
 RNG:     Mersenne-Twister 
 Normal:  Inversion 
 Sample:  Rounding 
 
locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] BiocParallel_1.20.1      TreeQTL_2.0              qvalue_2.18.0            data.table_1.12.8       
 [5] edgeR_3.28.1             factoextra_1.0.6         forcats_0.5.0            stringr_1.4.0           
 [9] dplyr_0.8.5              purrr_0.3.3              readr_1.3.1              tidyr_1.0.2             
[13] tibble_3.0.0             tidyverse_1.3.0          variancePartition_1.16.1 Biobase_2.46.0          
[17] BiocGenerics_0.32.0      scales_1.1.0             foreach_1.5.0            PCAtools_1.2.0          
[21] cowplot_1.0.0            lattice_0.20-38          reshape2_1.4.3           ggrepel_0.8.2           
[25] ggplot2_3.3.0            limma_3.42.2            

loaded via a namespace (and not attached):
  [1] snow_0.4-3                  readxl_1.3.1                backports_1.1.5            
  [4] circlize_0.4.8              fastmatch_1.1-0             plyr_1.8.6                 
  [7] ConsensusClusterPlus_1.50.0 splines_3.6.3               GenomeInfoDb_1.22.1        
 [10] digest_0.6.25               GOSemSim_2.12.1             GO.db_3.10.0               
 [13] lmerTest_3.1-1              gdata_2.18.0                fansi_0.4.1                
 [16] magrittr_1.5                memoise_1.1.0               cluster_2.1.0              
 [19] doParallel_1.0.15           ComplexHeatmap_2.2.0        Nozzle.R1_1.1-1            
 [22] modelr_0.1.6                matrixStats_0.56.0          prettyunits_1.1.1          
 [25] colorspace_1.4-1            blob_1.2.1                  rvest_0.3.5                
 [28] haven_2.2.0                 crayon_1.3.4                RCurl_1.98-1.1             
 [31] jsonlite_1.6.1              lme4_1.1-21                 survival_3.1-8             
 [34] iterators_1.0.12            glue_1.3.2                  gtable_0.3.0               
 [37] zlibbioc_1.32.0             XVector_0.26.0              GetoptLong_0.1.8           
 [40] DelayedArray_0.12.2         BiocSingular_1.2.2          shape_1.4.4                
 [43] DOSE_3.12.0                 DBI_1.1.0                   Rcpp_1.0.4                 
 [46] progress_1.2.2              lasso2_1.2-20               clue_0.3-57                
 [49] dqrng_0.2.1                 bit_1.1-15.2                rsvd_1.0.3                 
 [52] Formula_1.2-3               stats4_3.6.3                httr_1.4.1                 
 [55] fgsea_1.12.0                gplots_3.0.3                RColorBrewer_1.1-2         
 [58] acepack_1.4.1               ellipsis_0.3.0              pkgconfig_2.0.3            
 [61] farver_2.0.3                dbplyr_1.4.2                locfit_1.5-9.4             
 [64] tidyselect_1.0.0            labeling_0.3                rlang_0.4.5                
 [67] AnnotationDbi_1.48.0        munsell_0.5.0               cellranger_1.1.0           
 [70] tools_3.6.3                 cli_2.0.2                   generics_0.0.2             
 [73] RSQLite_2.2.0               broom_0.5.5                 yaml_2.2.1                 
 [76] bit64_0.9-7                 fs_1.4.0                    caTools_1.18.0             
 [79] nlme_3.1-144                DO.db_2.9                   xml2_1.3.0                 
 [82] compiler_3.6.3              pbkrtest_0.4-8.6            rstudioapi_0.11            
 [85] png_0.1-7                   reprex_0.3.0                statmod_1.4.34             
 [88] stringi_1.4.6               Matrix_1.2-18               nloptr_1.2.2.1             
 [91] vctrs_0.2.4                 pillar_1.4.3                lifecycle_0.2.0            
 [94] BiocManager_1.30.10         GlobalOptions_0.1.1         bitops_1.0-6               
 [97] irlba_2.3.3                 GenomicRanges_1.38.0        colorRamps_2.3             
[100] R6_2.4.1                    KernSmooth_2.23-16          gridExtra_2.3              
[103] IRanges_2.20.2              codetools_0.2-16            boot_1.3-24                
[106] MASS_7.3-51.5               gtools_3.8.2                assertthat_0.2.1           
[109] SummarizedExperiment_1.16.1 rjson_0.2.20                withr_2.1.2                
[112] S4Vectors_0.24.3            GenomeInfoDbData_1.2.2      hms_0.5.3                  
[115] grid_3.6.3                  minqa_1.2.4                 DelayedMatrixStats_1.8.0   
[118] logging_0.10-108            numDeriv_2016.8-1.1         base64enc_0.1-3            
[121] lubridate_1.7.4     
```

sessionInfo() on work cluster:
```
R version 3.6.3 (2020-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/libblas/libblas.so.3.6.0
LAPACK: /usr/lib/lapack/liblapack.so.3.6.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] BiocParallel_1.20.1      edgeR_3.28.1             variancePartition_1.16.1
[4] Biobase_2.46.0           BiocGenerics_0.32.0      scales_1.1.0            
[7] foreach_1.4.8            limma_3.42.2             ggplot2_3.3.0           

loaded via a namespace (and not attached):
 [1] progress_1.2.2      gtools_3.8.1        tidyselect_1.0.0   
 [4] locfit_1.5-9.1      purrr_0.3.3         reshape2_1.4.3     
 [7] splines_3.6.3       lmerTest_3.1-1      lattice_0.20-40    
[10] colorspace_1.4-1    vctrs_0.2.4         snow_0.4-3         
[13] rlang_0.4.5         pillar_1.4.3        nloptr_1.2.2.1     
[16] glue_1.3.2          withr_2.1.2         lifecycle_0.2.0    
[19] plyr_1.8.6          stringr_1.4.0       munsell_0.5.0      
[22] gtable_0.3.0        caTools_1.18.0      codetools_0.2-16   
[25] doParallel_1.0.15   pbkrtest_0.4-8.6    Rcpp_1.0.4         
[28] KernSmooth_2.23-16  gdata_2.18.0        lme4_1.1-21        
[31] gplots_3.0.3        hms_0.5.3           stringi_1.4.6      
[34] dplyr_0.8.5         numDeriv_2016.8-1.1 grid_3.6.3         
[37] bitops_1.0-6        tools_3.6.3         magrittr_1.5       
[40] tibble_2.1.3        crayon_1.3.4        pkgconfig_2.0.3    
[43] MASS_7.3-51.5       Matrix_1.2-18       prettyunits_1.1.1  
[46] assertthat_0.2.1    minqa_1.2.4         iterators_1.0.12   
[49] R6_2.4.1            boot_1.3-24         colorRamps_2.3     
[52] nlme_3.1-145        compiler_3.6.3 
```



Thanks,
Rachel

**Comment by raungar** [comment]
Update: restarting R solved this for me.

**Comment by GabrielHoffman** [comment]
Hi Rachel, 
I missed this bug report before because I got it at the same time as the other one.

Since I'm not able to reproduce this error, I have a free few

- Do you get this error consistently or does it seem random even with the same `set.seed()` value?
- Does it every occur in a fresh R session?
- Does it still occur of you only use 1 thread by running:
```r
library(BiocParallel)
register(SerialParam())
```
- Can you reproduce the error and send me the result of `bpparam()`

**Comment by raungar** [comment]
Hi-

No worries! Unfortunately, I cannot replicate this error, ever since restarting R it somehow fixed this issue. I am now able to use it as normal. I'll let you know if I see it again and follow this to document it for you.

Thanks,
Rachel

**Comment by GabrielHoffman** [comment]
Hi Rachel, 
I am able to get the error, but I'm not sure if it is in the same context as you.
```r
library(variancePartition)
data(varPartData)

form <- ~ (1|Individual) + (1|Tissue)

fit = dream( geneExpr[1:10,], form, info)
```
This gives: `Error in r[cbind(1L:p, 1L:p)] <- 1 : subscript out of bounds`

This is happening because `dream()` is trying to evaluate a hypothesis test, but there are no fixed effects to test in this formula.

I have fixed this in v1.17.10 so that it now says: 
```r
Warning message:
In .standard_transform(ret) :
  No testable fixed effects were included in the model.
  Running topTable() will fail.
```
Not ideal, but at least it fails more gracefully than before.
